tests/test_encoding.py

```diff
@@ -0,0 +1,17 @@
+"""Testing the parse_file function."""
+
+import pytest
+
+from bibtexparser import parse_file, writer
+
+
+def test_gbk():
+    library = parse_file("tests/gbk_test.bib", encoding="gbk")
+    lines = writer.write(library).splitlines()
```

Comment:
I would suggest not making the assertion on the "re-written" bibtex, but on the parsed entities instead.

e.g. `assert library.entries[0]["author"] == ...`

That way, we can keep the test independent of the writer.